tests/MongoDB.Driver.Tests/Specifications/crud/FindOneAndUpdateTest.cs

```diff
@@ -36,7 +36,14 @@ protected override bool TrySetArgument(string name, BsonValue value)
                     _filter = (BsonDocument)value;
                     return true;
                 case "update":
-                    _update = (BsonDocument)value;
+                    if (value.IsBsonArray)
+                    {
+                        _updatePipeline = (BsonArray)value;
+                    }
+                    else
+                    {
+                        _update = (BsonDocument) value;
+                    }
```

Comment:
If _update is declared as BsonValue as I suggest above lines 39-46 can simply be:

    _update = value;